src/workerd/io/io-context.c++

```diff
@@ -1164,6 +1159,17 @@ void IoContext::runImpl(Runnable& runnable,
       }
     }
   });
+
+  if (!isCurrentNull()) {
+    KJ_LOG(ERROR, "IoContext not-null before running PumpMessageLoop()");
+  } else {
+    worker->runInLockScope(lockType, [&](Worker::Lock& lock) {
+        jsg::Lock& js = lock;
+        auto& system = const_cast<jsg::V8System&>(js.getV8System());
+        KJ_DBG(js.v8Isolate);
+        while (v8::platform::PumpMessageLoop(&system.getDefaultPlatform(), js.v8Isolate, v8::platform::MessageLoopBehavior::kDoNotWait)) {}
```

Comment:
Is this actually the right time to call PumpMessageLoop -- after every single `ctx.run()`?

Arguably it might be preferable if we invoked `PumpMessageLoop` asynchronously later on when we're not actively responding to a request.

Reply:
I agree that we likely need to find a better place for this. We can end up running ctx.run quite a few times during a request.